app/services/force/event_subscriber_translate_service.rb

```diff
@@ -72,30 +69,47 @@ def setup_faye_client
       @faye_client.set_header('Authorization', "OAuth #{credentials.access_token}")
     end
 
+    def setup_translation_service
+      @translation_service = GoogleTranslationService.new
+    end
+
     def subscribe_to_listing_updates
       @faye_client.subscribe('/data/Listing__ChangeEvent') do |platform_event|
         logger(
           'New Salesforce event via /data/Listing__ChangeEvent: ' \
           "#{platform_event.inspect}",
         )
         event = parse_event(platform_event)
-        translate_and_log_event(event)
+        translate_and_cache(event)
       end
     end
 
-    def translate_and_log_event(event)
+    def translate_and_cache(event)
       translations = translate_event_values(event.updated_values)
-      @translation_service.cache_listing_translations(
+      logger(
+        "Event Translations: #{translations.inspect}",
+      )
+      if translations.empty?
+        logger(
+          "No translations for event: #{event.inspect}",
+        )
+        return []
+      end
+
+      cached_response = @translation_service.cache_listing_translations(
         event.listing_id,
         event.updated_values.keys,
         translations,
       )
       logger(
-        "Event Translations: #{translations.inspect}",
+        "Event Translations: #{cached_response.inspect}",
       )
     end
 
     def translate_event_values(values)
+      logger("Translating values: #{values.inspect}")
+      return [] unless values&.values
```

Comment:
is this how we make sure it still runs, by making sure it doesn't hit that error? What if it hits another unexpected error?

Reply:
TODO: look into automatic retries in future ticket